subsys/bluetooth/host/adv.c

```diff
@@ -1373,7 +1373,7 @@ static void adv_timeout(struct k_work *work)
 #else
 	err = bt_le_adv_stop();
 #endif
-	__ASSERT(err != 0, "Limited Advertising timeout reached, "
+	__ASSERT(err == 0, "Limited Advertising timeout reached, "
```

Comment:
This looks OK, but I'm seeing something suspicious in the lines above.

```
	if (atomic_test_bit(adv->flags, BT_ADV_EXT_ADV)) {
		err = bt_le_ext_adv_stop(adv);
	}
```

Why isn't the advertising set stopped if it is not configured as an extended advertiser?

Reply:
@sjanc I didn't find Krzyztof to tag, but do you know why it is this way?

Reply:
I think there are more issues with original PR...

1) bt_le_ext_adv_stop() and bt_le_adv_stop() should also cancel delayed work ie. if called by application before timeout is reached
2) delayed work should be cancelled when set terminated ie due to connection or parameters set via BT_LE_EXT_ADV_START_PARAM
3) if advertising is terminated, how application will know about this? should "sent" callback be called? what about legacy advertising (with ext_adv disabled in kconfig)?

Reply:
I found this problem in the PTS test, but I can't pass even after modifying assert.

Reply:
Thanks for checking, created PR to update that: https://github.com/zephyrproject-rtos/zephyr/pull/37439

Reply:
@sjanc Forgot to update, so didn't see your comment.

In that case the assert has to be removed, since we are now getting into corner cases with who disables first, and canceling is not guaranteed to succeed for all cases.

Reply:
> I found this problem in the PTS test, but I can't pass even after modifying assert.

Maybe because you stop the advertiser, and then the adv-timeout runs, as sjanc said.

Reply:
@sjanc I created https://github.com/zephyrproject-rtos/zephyr/issues/37467 to track this.